src/media/local-stream.ts

```diff
@@ -283,6 +283,36 @@ abstract class _LocalStream extends Stream {
     return this.effects;
   }
 
+  /**
+   * Method to serialize data about input, output streams
+   * and also effects from LocalStream.
+   *
+   * @returns - A JSON-compatible object representation with data from LocalStream.
+   */
+  toJSON() {
+    return {
+      muted: this.muted,
+      enabled: this.inputTrack.enabled,
+      label: this.label,
+      readyState: this.readyState,
+      inputStream: {
+        active: this.inputStream.active,
+        id: this.inputStream.id,
```

Comment:
...and add `enabled: this.inputTrack.enabled` and `muted: this.inputTrack.muted` here. This way, we can tell the `enabled` and `muted` state of the input track apart from the overall `muted` state of the `LocalStream`.

Reply:
@brycetham updated this part, thanks)